feature: rich Tefenua data graphql

yhru opened the pull request.

yhru committed:
feat(GraphQL): Enrichit l'API TeFenua avec les données de zone détaillées

yhru committed:
fix: reload schema graphql

maatinito commented on app/graphql/types/champs/te_fenua_champ_type.rb:
```diff
@@ -5,9 +5,39 @@ class TeFenuaChampType < Types::BaseObject
     implements Types::ChampType
 
     field :geo_areas, [Types::GeoAreaType], null: false
+    field :parcelles, [Types::GeoAreaType], null: false
```
cf nouvelles specs : à supprimer

maatinito commented on app/graphql/types/champs/te_fenua_champ_type.rb:
```diff
@@ -5,9 +5,39 @@ class TeFenuaChampType < Types::BaseObject
     def geo_areas
       Loaders::Association.for(Champs::TeFenuaChamp, :geo_areas).load(object)
     end
+
```
à supprimer (cf specs)

maatinito commented on app/graphql/types/geo_areas/parcelle_cadastrale_type.rb:
```diff
@@ -4,14 +4,18 @@ module Types::GeoAreas
-    field :surface, String, null: false
-    field :prefixe, String, null: false
-    field :commune, String, null: false
+    field :numero, String, null: true
```
pourquoi ca peut etre nul en Pf ?

maatinito commented on app/models/champs/te_fenua_champ.rb:
```diff
@@ -1,20 +1,23 @@
 
 class Champs::TeFenuaChamp < Champ
+  # Default map location for French Polynesia
+  DEFAULT_LON = -149.5665
```
Je n'ai pas compris à quoi cela servait ?

maatinito commented on app/models/champs/te_fenua_champ.rb:
```diff
@@ -128,12 +131,38 @@ def sync_geo_areas_from_value
+    end
+
+    # Handle legacy :positions format for markers
+    if marker? && parsed_data[:positions].is_a?(Array)
```
C'est quoi cette partie legacy ?

maatinito commented on app/models/geo_area.rb:
```diff
@@ -190,16 +191,23 @@ def surface
     end
+
+    # For TeFenua data, extract surface from info_texte if not available in standard fields
+    if api_surface.blank? && properties['info_texte'].present?
```
Modifions à la source : on a déjà à la source une manière de calculer la surface, appliquons la sur toutes les features ?

maatinito reviewed (commented):
Dune manière générale, lire le remplacement des specs et repartir de zéro. 
Avoir les attributs cadastre, zones_manuelles, etc... ca ne fait que dupliquer les geos_areas. 
Et si je veux une requete generique, je dois demander les 5 attributs (parcelles, batiments..., geoareas) et j'aurai forcément les elements en double dans geo_areas, et d'autres elements vides. 
<img width="689" height="792" alt="image" src="https://github.com/user-attachments/assets/9c6906e8-80f8-4551-bd6b-598cdf3290fe" />

yhru committed:
feat: new specs for graphql enriched data

yhru committed:
chore: linter

yhru committed:
feat: more spec for the feature

yhru committed:
chore: clean

yhru committed:
fix: spec

maatinito commented:
Hello, 
Merci pour les modifications. On est parfaitement dans l'esprit Graphql, ca fonctionne bien :-)
Voici quelques retours: 
1. Parcelles cadastralles: La surface est toujours vide. 
    <img width="404" height="604" alt="image" src="https://github.com/user-attachments/assets/514708f0-7e08-4809-b601-496141cd9816" />
    * c'est quoi la différence entre prefixe et section ? Est-ce que ca n'est pas la meme chose ?
2. Batiments :
    <img width="392" height="726" alt="image" src="https://github.com/user-attachments/assets/209faaac-7a67-49e6-a8a0-926397538f20" />
    * est-ce que nom et description c'est pas la meme chose ? 
    * pourquoi surfaceCalculée est toujours vide
3. Marqueurs
    <img width="470" height="451" alt="image" src="https://github.com/user-attachments/assets/7eec952e-055b-4988-aa86-77cd1674192d" />
    * Pourquoi la description est vide ? 
4.  Zones
    <img width="407" height="504" alt="image" src="https://github.com/user-attachments/assets/2c3cd862-9d33-4aab-8cf0-19cc3a031eef" />
    * Pourquoi avoir une surfaceCalculée et une surface ? Et pas simplement Surface ? 
5. Selection_utilisateur 
    <img width="436" height="520" alt="image" src="https://github.com/user-attachments/assets/bca00efe-330b-4ec7-ba27-ec56537c17ff" />
    * qu'est ce qui motive le fait d'avoir une selection_utilisateur dans le schema graphql alors qu'il y a cadastre, batiment, zone, marqueurs ?

yhru committed:
fix: change according to review

yhru committed:
fix: spec

maatinito committed:
fix: ajoute implements GeoAreaType aux types héritants pour génération GraphQL

- ZoneType et BatimentType doivent explicitement implémenter GeoAreaType
- Restaure le champ surfaceCalculee perdu lors du rebase
- Régénère les schémas GraphQL avec tous les types corrects
